<a href="https://colab.research.google.com/github/SurajLahariya/DS_learning/blob/main/Assignment_3_(_Q_no_4_)_Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
from scipy import stats
from scipy.stats import norm

In [2]:
url = 'https://raw.githubusercontent.com/SurajLahariya/DS_learning/main/Costomer%2BOrderForm.csv'

In [4]:
data = pd.read_csv(url)

In [5]:
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [6]:
Phillippines = data.Phillippines.value_counts()
Indonesia = data.Indonesia.value_counts()
Malta = data.Malta.value_counts()
India = data.India.value_counts()

In [7]:
Phillippines

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [8]:
Indonesia

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [9]:
Malta

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [10]:
India

Error Free    280
Defective      20
Name: India, dtype: int64

In [11]:
Data1 = pd.DataFrame(index = ['Error Free','Defective'],
                  data = {'Phillippines':[271,29],
                          'Indonesia':[267,33],
                          'Malta':[269,31],
                          'India':[280,20]})
Data1

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [12]:
from scipy.stats import chi2_contingency
p_value = stats.chi2_contingency(Data1)

In [13]:
p_value[1]

0.2771020991233135

###p-value (0.277)> α (0.05), So we can't reject the null hypothesis.
Percentages of Defective across all the centres is same.